src/utils/mod.rs

```diff
@@ -1377,12 +1376,115 @@ impl Default for PcrSelectSize {
 /// number of PCR required by the platform-specific
 /// specification with which the TPM is compliant or by the implementer if
 /// not adhering to a platform-specific specification.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
 pub struct PcrSelections {
     size_of_select: PcrSelectSize,
     items: HashMap<HashingAlgorithm, BitFlags<PcrSlot>>,
 }
 
+impl PcrSelections {
+    /// Function for retrieiving the number of banks in the selection
+    pub fn len(&self) -> usize {
+        self.items.len()
+    }
+
+    /// Returns true if the selection is empty.
+    pub fn is_empty(&self) -> bool {
+        self.items.is_empty()
+    }
+
+    /// Returns the the elements present in `self` but not in `other` or None
+    /// if they are equal.
```

Comment:
I'd lean towards a `subtract` method to mimick, for example, the [`split_off`](https://doc.rust-lang.org/std/collections/struct.LinkedList.html#method.split_off) in `LinkedList`. Alternatively an iterator that only points to the different elements could be returned. But I agree, it depends on the use case.

Reply:
The use-case for the method it is to get the remaining items when you do I pcr_read for example.

In the case  that the you have a PcrSelections that contains more then 16 items then it will not be able to get all of them in one read. So I was thinking you sould be able to do something like
```rust
let mut remaining = Some(pcr_selections);
while remaining.is_some() {
   let (_update_counter, read_pcr_selections, data) = ctx.pcr_read(remaining.unwrap());
   remaining = remaining.map(|x| x.difference(read_pcr_selections);
}
```

But in order to be able to do that properly. There should probably be some kind of merge funtion for the PcrData struct.

Reply:
Could this:
```rust
remaining = remaining.map(|x| x.difference(read_pcr_selections));
```
be replaced with something like (as an idea)
```rust
remaining.substract(read_pcr_selections);
```

Reply:
yeah absolutly something like:
```rust
let mut remaining = pcr_selections;
while !remaining.is_empty() {
    let (update_counter, read_pcr_selections, pcr_data) = ctx.read_pcr(remaining).unwrap();
    remaining.subtract(read_pcr_selections);
}
```
I think that works just as well.

Reply:
Fixed